trio/_core/_local.py

```diff
@@ -28,44 +39,44 @@ class RunVar(metaclass=Final):
 
     """
 
-    _NO_DEFAULT = object()
-    _name = attr.ib()
-    _default = attr.ib(default=_NO_DEFAULT)
+    _name: str = attr.ib()
+    _default: T | type[_NoValue] = attr.ib(default=_NoValue)
 
-    def get(self, default=_NO_DEFAULT):
+    def get(self, default: T | type[_NoValue] = _NoValue) -> T:
         """Gets the value of this :class:`RunVar` for the current run call."""
         try:
-            return _run.GLOBAL_RUN_CONTEXT.runner._locals[self]
+            # not typed yet
+            return _run.GLOBAL_RUN_CONTEXT.runner._locals[self]  # type: ignore[return-value, index]
         except AttributeError:
             raise RuntimeError("Cannot be used outside of a run context") from None
         except KeyError:
             # contextvars consistency
-            if default is not self._NO_DEFAULT:
-                return default
+            if default is not _NoValue:
+                return default  # type: ignore[return-value]
 
-            if self._default is not self._NO_DEFAULT:
-                return self._default
+            if self._default is not _NoValue:
+                return self._default  # type: ignore[return-value]
 
             raise LookupError(self) from None
 
-    def set(self, value):
+    def set(self, value: T) -> RunVarToken[T]:
         """Sets the value of this :class:`RunVar` for this current run
         call.
 
         """
         try:
             old_value = self.get()
         except LookupError:
-            token = _RunVarToken.empty(self)
+            token = RunVarToken[T]._empty(self)
```

Comment:
Huh, I feel like mypy should infer `T`.

Reply:
I guess it doesn't backtrack from `return token` to determine `token: RunVarToken[T]`. Might be better to just add a variable annotation before the `try` block for it, then it should be able to infer.

Reply:
Something something inheritance seems to break the typevar (or maybe because one is `trio._socket.T` and one is `trio._util.T`), so line 72 needs an explicit `T` (otherwise I get `Incompatible types in assignment (expression has type "RunVarToken[T]", variable has type "RunVarToken[T]")`) and a variable annotation before the `try` doesn't help. But I could remove the one at line 70.